portality/bll/services/article.py

```diff
@@ -103,7 +105,8 @@ def _batch_contains_duplicates(self, articles):
         return False
 
     def create_article(self, article, account, duplicate_check=True, merge_duplicate=True,
-                       limit_to_account=True, add_journal_info=False, dry_run=False):
+                       limit_to_account=True, add_journal_info=False, dry_run=False, update=None):
```

Comment:
> * `update` as a new argument needs to go into the `argvalidate` call to validate the incoming type
> 
> * The docs on the arg say "The article that it is supposed to be an update to", but the argument itself is the article id.  Should we pass the id (in which case update the docs) or the actual article (in which case update the calling code).
> 
> * Perhaps a better name for the argument would be `replace_article`, to be super clear

done. Name changed to `update_article_id`, accepting id, doc updated, validated by: 
`{"arg" : update_article_id, "instance" : str, "allow_none" : False, "arg_name" : "update_article_id"}`

Reply:
> Also, at the moment, if `duplicate_check=True` and `merge_duplicate=True`, and the detected duplicate has the same id as the update (which I expect is almost every time this function runs), then `article.merge` will always run twice in this one function, with the same argument in both runs. We should make sure that if `article.merge` is already run, then it doesn't get run a second time.

The second run is unnecessary so I'm removing it. If `update_article_id` is not None then it will be checked against the duplicate found and replaced in `if duplicate_check:`